src/MongoDB.Driver/FindFluent.cs

```diff
@@ -59,16 +59,58 @@ public override FilterDefinition<TDocument> Filter
             return Project(projection);
         }
 
+        [Obsolete("Use CountDocuments instead.")]
         public override long Count(CancellationToken cancellationToken)
         {
             var options = CreateCountOptions();
-            return _collection.Count(_filter, options, cancellationToken);
+            if (_session == null)
+            {
+                return _collection.Count(_filter, options, cancellationToken);
+            }
+            else
+            {
+                return _collection.Count(_session, _filter, options, cancellationToken);
+            }
```

Comment:
This was a bug in the existing Count implementation. The tests for Count have been updated accordingly.